python/ray/tune/integration/mlflow.py

```diff
@@ -322,9 +338,13 @@ def __init__(self, config: Dict, *args, **kwargs):
                     "key in your `config` dict containing at "
                     "least a `experiment_name` or `experiment_id` "
                     "specification.")
-            # Name exists.
             experiment = self._mlflow.get_experiment_by_name(experiment_name)
-            experiment_id = experiment.experiment_id
+            if experiment is not None:
+                # Experiment with this name exists.
+                experiment_id = experiment.experiment_id
+            else:
+                raise ValueError("No experiment with the given experiment "
+                                 "name or id currently exists.")
```

Comment:
nit - consider printing the provided experiment_name in the error message

Reply:
Good point- fixed